dashboards-observability/public/components/synthetics/helpers/test_details_flyout.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    EuiButton,
+    EuiComboBox,
+    EuiComboBoxOptionOption,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiFlyoutBody,
+    EuiFlyoutFooter,
+    EuiFlyoutHeader,
+    EuiLink,
+    EuiMarkdownFormat,
+    EuiSpacer,
+    EuiText,
+    EuiTitle,
+  } from '@elastic/eui';
+  import { PPL_DOCUMENTATION_URL } from '../../../../common/constants/shared';
+  import _ from 'lodash';
+  import React, { useState } from 'react';
+  import { FlyoutContainers } from '../../common/flyout_containers';
+  import { Group1, Group2, Group3 } from '../../common/helpers/ppl_docs/groups';
+  import { overview } from '../../common/helpers/ppl_docs/overview';
+import { Plt } from '../../visualizations/plotly/plot';
+import { CONTENT_SIZE_FIELD, DOWNLOAD_SPEED_FIELD, DOWNLOAD_TIME_FIELD, REDIRECT_TIMING_FIELD, STARTTRANSFER_TIMING_FIELD, TEST_SUITE_NAME_FIELD, TIMING_FIELDS, URL_FIELD } from '../../../../common/constants/synthetics';
+  
+  type Props = {
+    testSuiteObject: Object | undefined;
+    closeFlyout: () => void;
+  };
+  
+  export const TestDetailsFlyout = ({ testSuiteObject, closeFlyout }: Props) => {
+    
+    // contains all the timing fields being shown
+    const allTimingFields = () => {
+      var retval = []
+      retval = TIMING_FIELDS.slice()
+      retval.push(DOWNLOAD_TIME_FIELD)
+      retval.unshift('Content Download')
+      return retval
+    }
+
+    // contains just the timing fields listed in TIMING_FIELDS
+    const allTimingValues = () => {
+      var retval = [];
+      for (var i = 0; i < TIMING_FIELDS.length; i++) {
+        retval.push(testSuiteObject[TIMING_FIELDS[i]]);
+      }
+      if (testSuiteObject[REDIRECT_TIMING_FIELD] == 0) {
+        var redirIndex = TIMING_FIELDS.indexOf(REDIRECT_TIMING_FIELD);
+        retval[redirIndex] = retval[redirIndex+1];
+      }
+      return retval;
+    }
+
+    // calculates every field's base and length to display
+    const bases_values = () => {
+      let bases = [];
+      let values = [];
+      let timingVals = allTimingValues();
+      let downloadTime = testSuiteObject[DOWNLOAD_TIME_FIELD]
+      let contentDownload = downloadTime - timingVals[0];
+      // content download bar
+      bases.push(timingVals[0])
+      values.push(contentDownload)
+      for (var i = 0; i < timingVals.length-1; i++) {
+        // have it shown as ttfb
+        if (TIMING_FIELDS[i] == STARTTRANSFER_TIMING_FIELD) {
```

Comment:
This could also be i = 0 I think??

Reply:
I have it like this just in case the order of the timing fields gets modified, so that this will reference the named field instead of the position of where the field is.